From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Sauve <paul@technove.co>
Date: Wed, 11 Nov 2020 19:31:14 -0600
Subject: [PATCH] Correctly tick goalSelector for inactive entities

Previously, inactive entities wouldn't ever have their goalSelector
ticked if they were inactive. This breaks things like tamed animals
teleporting back to the player if they're out of range.

diff --git a/src/main/java/net/minecraft/server/EntityInsentient.java b/src/main/java/net/minecraft/server/EntityInsentient.java
index 23007c9ccb4cfb9b8ba817ab5aaf7afbc3bb4aae..1311dca17742a3bb26609cde827639b094b45ac4 100644
--- a/src/main/java/net/minecraft/server/EntityInsentient.java
+++ b/src/main/java/net/minecraft/server/EntityInsentient.java
@@ -132,7 +132,11 @@ public abstract class EntityInsentient extends EntityLiving {
     @Override
     public void inactiveTick() {
         super.inactiveTick();
-        this.goalSelector.inactiveTick();
+        // AirplaneL start - actually tick goalSelector, sometimes
+        if (this.goalSelector.inactiveTick()) {
+            this.goalSelector.doTick();
+        }
+        // AirplaneL end
         if (this.targetSelector.inactiveTick()) {
             this.targetSelector.doTick();
         }
